x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchMaintenanceService.java

```diff
@@ -107,11 +111,17 @@ synchronized void scheduleNextCleanup() {
         }
     }
 
+    synchronized void stop() {
+        if (isCleanupRunning.compareAndSet(true, false)) {
+            if (cancellable != null && cancellable.isCancelled() == false) {
+                cancellable.cancel();
+            }
+        }
+    }
```

Comment:
I am curious about these changesL: the newly added tests uncovered some bug?

Reply:
Yes, if the index is deleted we close the service. I added a test for this `AsyncSearchActionIT#testRemoveAsyncIndex`. I can add a note in the commit/pr or open a separate pr if you prefer.